pkg/auth/oauth/external/google/google.go

```diff
@@ -1,113 +1,36 @@
 package google
 
 import (
-	"encoding/base64"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"strings"
-
-	"github.com/RangelReale/osincli"
-	"github.com/golang/glog"
-
-	authapi "github.com/openshift/origin/pkg/auth/api"
 	"github.com/openshift/origin/pkg/auth/oauth/external"
+	"github.com/openshift/origin/pkg/auth/oauth/external/openid"
 )
 
-const (
+var (
 	googleAuthorizeURL = "https://accounts.google.com/o/oauth2/auth"
-	googleTokenURL     = "https://accounts.google.com/o/oauth2/token"
-	googleOAuthScope   = "profile email"
+	googleTokenURL     = "https://www.googleapis.com/oauth2/v3/token"
+	googleUserInfoURL  = "https://www.googleapis.com/oauth2/v3/userinfo"
+	googleOAuthScopes  = []string{"openid", "email", "profile"}
 )
 
 type provider struct {
 	providerName, clientID, clientSecret string
 }
 
-func NewProvider(providerName, clientID, clientSecret string) external.Provider {
-	return provider{providerName, clientID, clientSecret}
-}
+func NewProvider(providerName, clientID, clientSecret string) (external.Provider, error) {
+	config := openid.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
 
-// NewConfig implements external/interfaces/Provider.NewConfig
-func (p provider) NewConfig() (*osincli.ClientConfig, error) {
-	config := &osincli.ClientConfig{
-		ClientId:                 p.clientID,
-		ClientSecret:             p.clientSecret,
-		ErrorsInStatusCode:       true,
-		SendClientSecretInParams: true,
-		AuthorizeUrl:             googleAuthorizeURL,
-		TokenUrl:                 googleTokenURL,
-		Scope:                    googleOAuthScope,
-	}
-	return config, nil
-}
+		AuthorizeURL: googleAuthorizeURL,
+		TokenURL:     googleTokenURL,
+		UserInfoURL:  googleUserInfoURL,
 
-// AddCustomParameters implements external/interfaces/Provider.AddCustomParameters
-func (p provider) AddCustomParameters(req *osincli.AuthorizeRequest) {
-	req.CustomParameters["include_granted_scopes"] = "true"
-	req.CustomParameters["access_type"] = "offline"
```

Comment:
In case you're wondering, Google changed their implementation to converge on the final OpenID Connect spec... hence the scope change to add "openid", the removal of these custom params, and the addition of the UserInfo fetch (to get more details like display name)